src/basic/nss-util.h

```diff
@@ -137,7 +137,7 @@ enum nss_status _nss_##module##_getpwnam_r(             \
                 struct passwd *pwd,                     \
                 char *buffer, size_t buflen,            \
                 int *errnop) _public_;                  \
-enum nss_status _nss_mymachines_getpwuid_r(             \
+enum nss_status _nss_##module##_getpwuid_r(             \
```

Comment:
so far we only had one NSS module that provided user database lookups (all the others only provided host name lookups), and that one module was called "mymachines", hence the macro was resolved to the same name always...